planning/autoware_static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp

```diff
@@ -188,4 +217,140 @@ std::vector<TrajectoryPoint> OptimizationTrajectoryBasedCenterline::optimize_tra
+
+  const std::vector<double> goal_pose_input =
+    autoware::universe_utils::getOrDeclareParameter<std::vector<double>>(node, "end_pose");
+  const Pose goal_pose = utils::create_pose(
+    goal_pose_input[0], goal_pose_input[1], goal_pose_input[2], goal_pose_input[3],
+    goal_pose_input[4], goal_pose_input[5], goal_pose_input[6]);
+  return {start_pose, goal_pose};
+}
+
+std::shared_ptr<autoware_planning_msgs::msg::LaneletRoute>
+OptimizationTrajectoryBasedCenterline::create_route(
+  std::shared_ptr<RouteHandler> & route_handler_ptr, const geometry_msgs::msg::Pose & start_pose,
+  const geometry_msgs::msg::Pose & goal_pose) const
+{
+  // create route_ptr
+  std::shared_ptr<autoware_planning_msgs::msg::LaneletRoute> route_ptr =
+    std::make_shared<autoware_planning_msgs::msg::LaneletRoute>();
+
+  lanelet::ConstLanelets all_lanelets;
+  lanelet::ConstLanelets path_lanelets;
+  if (!route_handler_ptr->planPathLaneletsBetweenCheckpoints(
+        start_pose, goal_pose, &path_lanelets)) { /* do nothing */
+  }
+
+  for (const auto & lane : path_lanelets) {
+    if (!all_lanelets.empty() && lane.id() == all_lanelets.back().id()) continue;
+    all_lanelets.push_back(lane);
+  }
+
+  route_handler_ptr->setRouteLanelets(all_lanelets);
+  std::vector<autoware_planning_msgs::msg::LaneletSegment> route_sections =
+    route_handler_ptr->createMapSegments(all_lanelets);
+
+  // set necessary data
+  route_ptr->header.frame_id = "map";
+  route_ptr->set__start_pose(start_pose);
+  route_ptr->set__goal_pose(goal_pose);
+  route_ptr->set__segments(route_sections);
+  return route_ptr;
+}
+
+std::shared_ptr<autoware::path_generator::PathGenerator>
+OptimizationTrajectoryBasedCenterline::create_path_generator_node(
+  const geometry_msgs::msg::Pose current_pose, LaneletMapBin::ConstSharedPtr & map_bin_ptr,
+  std::shared_ptr<autoware_planning_msgs::msg::LaneletRoute> route_ptr) const
+{
+  const auto path_generator_node =
+    std::make_shared<autoware::path_generator::PathGenerator>(create_node_options());
+
+  // set data in path_generator
+  autoware::path_generator::PathGenerator::InputData path_generator_input;
+  path_generator_input.lanelet_map_bin_ptr = map_bin_ptr;
+  path_generator_input.odometry_ptr = utils::convert_to_odometry(current_pose);
+  path_generator_input.route_ptr = route_ptr;
+  path_generator_node->set_planner_data(path_generator_input);
+
+  return path_generator_node;
+}
+
+std::shared_ptr<autoware::behavior_path_planner::PlannerData>
+OptimizationTrajectoryBasedCenterline::create_behavior_path_planner_data(
+  std::shared_ptr<RouteHandler> & route_handler_ptr) const
+{
+  // create planner_data
+  std::shared_ptr<autoware::behavior_path_planner::PlannerData> planner_data_ =
+    std::make_shared<autoware::behavior_path_planner::PlannerData>();
+  planner_data_->route_handler = route_handler_ptr;
+  planner_data_->parameters.refine_goal_search_radius_range = refine_goal_search_radius_range;
```

Comment:
Define refine_goal_search_radius_range variable here with `getOrDeclareParameter` and remove the member variable.